tests/src/python/test_selective_masking.py

```diff
@@ -1095,13 +1191,130 @@ def test_layout_export_2_sources_masking(self):
         # and mask other symbol layers underneath
         fmt.mask().setMaskedSymbolLayers([
             # the black part of roads
-            QgsSymbolLayerReference(self.lines_layer.id(), QgsSymbolLayerId("", 0))])
+            self.get_symbollayer_ref(self.lines_layer, "", [0])])
 
         label_settings.setFormat(fmt)
         self.polys_layer.labeling().setSettings(label_settings)
 
         self.check_layout_export("layout_export_2_sources_masking", 0)
 
+    def test_raster_line_pattern_fill(self):
+        """
+        Test raster rendering and masking when a line pattern fill symbol layer is involved
+        """
+        self.assertTrue(QgsProject.instance().read(os.path.join(unitTestDataPath(), "selective_masking_fill_symbollayer.qgz")))
+
+        layer = QgsProject.instance().mapLayersByName('line_pattern_fill')[0]
+        self.assertTrue(layer)
+
+        map_settings = QgsMapSettings()
+        crs = QgsCoordinateReferenceSystem('epsg:4326')
+        extent = QgsRectangle(-0.972, -1.966, 1.58, 1.806)
+        map_settings.setBackgroundColor(QColor(152, 219, 249))
+        map_settings.setOutputSize(QSize(420, 280))
+        map_settings.setOutputDpi(72)
+        map_settings.setFlag(QgsMapSettings.Antialiasing, True)
+        map_settings.setFlag(QgsMapSettings.UseAdvancedEffects, False)
+        map_settings.setDestinationCrs(crs)
+        map_settings.setExtent(extent)
+
+        map_settings.setLayers([layer])
+
+        self.check_renderings(map_settings, "line_pattern_fill")
+
+    def test_vector_line_pattern_fill(self):
+        """
+        Test vector rendering and masking when a line pattern fill symbol layer is involved
+        """
+        self.assertTrue(QgsProject.instance().read(os.path.join(unitTestDataPath(), "selective_masking_fill_symbollayer.qgz")))
+
+        layer = QgsProject.instance().mapLayersByName('line_pattern_fill')[0]
+        self.assertTrue(layer)
+
+        map_settings = QgsMapSettings()
+        crs = QgsCoordinateReferenceSystem('epsg:4326')
+        extent = QgsRectangle(-1.0073971192118132, -0.7875782447946843, 0.87882587741257345, 0.51640826470600099)
+        map_settings.setBackgroundColor(QColor(152, 219, 249))
+        map_settings.setOutputSize(QSize(420, 280))
+        map_settings.setOutputDpi(72)
+        map_settings.setFlag(QgsMapSettings.Antialiasing, True)
+        map_settings.setFlag(QgsMapSettings.UseAdvancedEffects, False)
+        map_settings.setDestinationCrs(crs)
+
+        map_settings.setLayers([layer])
+
+        self.check_layout_export("layout_export_line_pattern_fill", 0, [layer], extent=extent)
```

Comment:
The text in this image is very small and non-bold, so this test won't be stable across platforms. Can you make it bigger and bold, and ensure that it uses the standard QGIS test font too?

Reply:
Done